src/main/scala/s3/website/model/Config.scala

```diff
@@ -37,25 +37,22 @@ case class Config(
 object Config {
 
   def awsCredentials(config: Config): AWSCredentialsProvider = {
-    val credentialsFromConfigFile =
-      if (config.session_token.isEmpty) {
+    val credentialsFromConfigFile: Option[AWSStaticCredentialsProvider] =
+      if (config.s3_id.isEmpty) {
+         None
+      } else if (config.session_token.isEmpty) {
         for {
           s3_id <- config.s3_id
           s3_secret <- config.s3_secret
-        } yield new BasicAWSCredentials(s3_id, s3_secret)
+        } yield new AWSStaticCredentialsProvider(new BasicAWSCredentials(s3_id, s3_secret))
       } else {
         for {
           s3_id <- config.s3_id
           s3_secret <- config.s3_secret
           session_token <- config.session_token
-        } yield new BasicSessionCredentials(s3_id, s3_secret, session_token)
+        } yield new AWSStaticCredentialsProvider(new BasicSessionCredentials(s3_id, s3_secret, session_token))
       }
-    credentialsFromConfigFile.fold(new DefaultAWSCredentialsProviderChain: AWSCredentialsProvider)(credentials =>
-      new AWSCredentialsProvider {
-        def getCredentials = credentials
-        def refresh() = {}
-      }
-    )
+    new AWSCredentialsProviderChain(List(credentialsFromConfigFile, Some(new DefaultAWSCredentialsProviderChain)).flatten)
```

Comment:
Perhaps the following is more idiomatic Scala, because it leverages the ideas behind `Option`:

```scala
credentialsFromConfigFile getOrElse new DefaultAWSCredentialsProviderChain
```